src/charm.py

```diff
@@ -268,12 +268,20 @@ def scrape_jobs(self) -> List[Dict[str, Any]]:
             return [job]
 
         external_url = urlparse(self.ingress_per_unit.url)
-        return [
-            {
-                "metrics_path": f"{external_url.path}/metrics",
-                "static_configs": [{"targets": [f"{external_url.hostname}:{external_url.port}"]}],
-            }
-        ]
+
+        metrics_path = f"{external_url.path if external_url.path else ''}/metrics"
```

Comment:
Because of
```python
>>> urlparse("scheme://hostname").path
''
>>> urlparse("scheme://hostname/").path
'/'
```
we probably need:
```suggestion
        metrics_path = f"{external_url.path.rstrip("/")}/metrics"
```